src/ngx_http_lua_balancer.c

```diff
@@ -480,6 +480,24 @@ ngx_http_lua_ffi_balancer_set_current_peer(ngx_http_request_t *r,
         return NGX_ERROR;
     }
 
+    if (ssl == 1) {
+#if (NGX_HTTP_SSL)
+        if (u->conf->ssl) {
+            u->ssl = 1;
+        } else {
+            *err = "proxy_pass http protocol only";
+            return NGX_ERROR;
+        }
+#else
+        *err = "https protocol requires SSL support";
+        return NGX_ERROR;
+#endif
+    } else if (ssl == 0) {
+        u->ssl = 0;
+    } else {
+        //by default
```

Comment:
Style: C++ style line comments should NEVER be used. You cannot find a single `//` comment in the NGINX core, can you? ;)